src/Handling/SpellController.php

```diff
@@ -169,15 +160,18 @@ public function index()
         }
 
         // Check params and request type
-        if (!Director::is_ajax() || empty($data['method']) || empty($data['params']) || count($data['params']) < 2) {
+        if (!Director::is_ajax() || empty($data['method']) || empty($data['lang'])) {
             return $this->error(_t(__CLASS__ . '.InvalidRequest', 'Invalid request'), 400);
         }
 
         // Check locale
-        $params = $data['params'];
-        $locale = $params[0];
+        $locale = $data['lang'];
         if (!in_array($locale, self::get_locales())) {
-            return $this->error(_t(__CLASS__ . '.InvalidLocale', 'Not supported locale'), 400);
+            // Check if the locale is actually a language
```

Comment:
It may be safer style to roll-back this unrolled loop and save doing the same thing twice.

Reply:
Yep, good idea. I've updated and pushed this.